### PR TITLE
4249: remove code from user migration script for importing credentials

### PR DESCRIPTION
**Pull request by collinpreston**
#### Pre-Flight checklist
- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/4250 & https://github.com/mitodl/open-discussions/issues/4249.

#### What's this PR do?

Updates the Keycloak user migration script:
1. Removes functionality for migrating the user's password.
2. Adds functionality to specify a Keycloak group that any migrated users will be added to.
3. Adds functionality to only migrated Django user records with an specified associated social-auth provider, and no associated social-auth record for the "ol-apps" provider.

Updates Keycloak user migration script documentation: https://docs.google.com/document/d/17tJ-C2EwWoSpJWZKjuhMVgsqGtyPH0IN9KakXvSKU0M/edit?pli=1#heading=h.ds1pmqw8dg34

#### How should this be manually tested?
The following scenarios should be repeatable.
1. All Django user records without an associated social-auth record for the "ol-apps" provider have a Keycloak account created when the script is run without the `filter_provider_name` argument.
2. All Django user records without an associated social-auth record for the "ol-apps" provider but must have an associated social-auth record for the "email" provider have a Keycloak account created when the script is run with the `--filter_provider_name=email` argument.  Any users without a social-auth record should not be created.
3. All Django user records without an associated social-auth record for the "ol-apps" provider have a Keycloak account created and are added to the Keycloak group "/test/test-child" when the script is run without the `filter_provider_name` argument and with `--keycloak_group_path=/test/test-child`.
4. Any user created in Keycloak via the script should not have any credential.

#### Where should the reviewer start?
Documentation.

#### Any background context you want to provide?
https://docs.google.com/spreadsheets/d/1gOCQLNk1hBMutYoZVZ0Qa3SWb6LHRO9sepXCxy7X2q4/edit#gid=1942862761

### CODE REVIEW

File: keycloak_user_export/management/commands/migrate_users_to_keycloak.py [hunk 1]
```diff
@@ -63,44 +77,57 @@ def add_arguments(self, parser):
...
+            password (str): The password associated with the username for a Keycloak realm administrator user.
             client_secret (str): The client secret associated with Keycloak's admin-cli client.
 
         Returns:
-            A new access_token for the administrator user for use with the Keycloak admin-cli client.
+            A new access_token (string) for the administrator user for use with the Keycloak admin-cli client.
         """
         url = f"{settings.KEYCLOAK_BASE_URL}/realms/{settings.KEYCLOAK_REALM_NAME}/protocol/openid-connect/token"
-        payload = f"{urlencode({'client_id': client_id})}&{urlencode({'username': email})}&{urlencode({'password': password})}&grant_type=password&{urlencode({'client_secret': client_secret})}&{urlencode({'scope': 'email openid'})}"
+        payload = f"{urlencode({'client_id': client_id})}&{urlencode({'username': username})}&{urlencode({'password': password})}&grant_type=password&{urlencode({'client_secret': client_secret})}&{urlencode({'scope': 'email openid'})}"
```
[the remark this answers is not in the document]
Comment (collinpreston) [reply keycloak_user_export/management/commands/migrate_users_to_keycloak.py:117]: Is this possible with inserting the '&' between each dictionary item?

[the remark this answers is not in the document]
Comment (rhysyngsun) [reply keycloak_user_export/management/commands/migrate_users_to_keycloak.py:117]: @collinpreston it automatically does that for you:
```python
In [4]: parse.urlencode(dict(a=1,b="http://my.url"))
Out[4]: 'a=1&b=http%3A%2F%2Fmy.url'
```

**Comment by rhysyngsun** [comment]
@ChristopherChudzicki do you happen to have a local keycloak instance setup you could test with? I've been trying for 2 days to get me local env to work to no avail. The code here itself looks good though.

**Review by rhysyngsun — APPROVED** [review]
LGTM as far as the code goes, I haven't been able to get Keycloak to cooperate locally so this approval is only on a basis of code review, but that's the best I can do.

**Pull request merged**